eZ/Bundle/EzPublishCoreBundle/Command/NormalizeImagesPathsCommand.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Bundle\EzPublishCoreBundle\Command;
+
+use Doctrine\DBAL\Driver\Connection;
+use eZ\Publish\Core\FieldType\Image\ImageStorage\Gateway as ImageStorageGateway;
+use eZ\Publish\Core\IO\FilePathNormalizerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @internal
+ */
+final class NormalizeImagesPathsCommand extends Command
+{
+    private const IMAGE_LIMIT = 100;
+    private const BEFORE_RUNNING_HINTS = <<<EOT
+<error>Before you continue:</error>
+- Make sure to back up your database.
+- Run this command in production environment using <info>--env=prod</info>
+- Manually clear SPI/HTTP cache after running this command.
+EOT;
+
+    /** @var \eZ\Publish\Core\FieldType\Image\ImageStorage\Gateway */
+    private $imageGateway;
+
+    /** @var \eZ\Publish\Core\IO\FilePathNormalizerInterface */
+    private $filePathNormalizer;
+
+    /** @var \Doctrine\DBAL\Driver\Connection */
+    private $connection;
+
+    public function __construct(
+        ImageStorageGateway $imageGateway,
+        FilePathNormalizerInterface $filePathNormalizer,
+        Connection $connection
+    ) {
+        parent::__construct();
+
+        $this->imageGateway = $imageGateway;
+        $this->filePathNormalizer = $filePathNormalizer;
+        $this->connection = $connection;
```

Comment:
https://symfony.com/doc/3.4/console/commands_as_services.html#lazy-loading

Not using `protected static $defaultName` causes command to be instantiated, along with it's services, when `bin/console list` is called.

Btw, will the correct `Connection` be inserted here, i.e. is it aware that there might be many, depending on siteaccess and whatnot?

Reply:
I remember that autowiring might mess up the connection, changed it to depend on `@ezpublish.persistence.connection` instead.